manifests/params.pp

```diff
@@ -27,26 +27,31 @@
     }
   }
 
-  case $::puppetversion {
-    /^3\.\d\.\d$/: {
+  if str2bool($::is_pe) {
+    #Assuming PE 3
+    $gem_provider = 'pe_gem'
+    $group = 'pe-puppet'
+    $user = 'pe-puppet'
+  } else {
+    if $::puppetversion and versioncmp($::puppetversion, '4.0.0') >= 0 {
+      if $::pe_server_version
```

Comment:
I would really like to avoid nesting conditionals this deeply. One approach I've taken in the past, when there were a bunch of similar settings in params.pp, is to set a hash of defaults, then use a case statement to set overrides, and then merge them. (example in https://github.com/basho-labs/puppet-riak/blob/master/manifests/params.pp)

My ability to reason about puppet code slows down a lot after one level of nested conditional statements. Can you think of a better way to handle it?